pkg/agent/evictionmanager/plugin/memory/rss_overuse.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/kubewharf/katalyst-core/pkg/util/process"
+	"github.com/kubewharf/katalyst-core/pkg/util/qos"
+)
+
+const (
+	EvictionPluginNameRssOveruse = "rss-overuse-eviction-plugin"
+
+	RssOveruseEvictionReason = "hit rss overuse policy, threshold is %.2f, current pod rss is %.2f, pod memory limit is %d"
+)
+
+func NewRssOveruseEvictionPlugin(_ *client.GenericClientSet, _ events.EventRecorder,
+	metaServer *metaserver.MetaServer, emitter metrics.MetricEmitter, conf *config.Configuration) plugin.EvictionPlugin {
+	return &RssOveruseEvictionPlugin{
+		StopControl:        process.NewStopControl(time.Time{}),
+		emitter:            emitter,
+		reclaimedPodFilter: conf.CheckReclaimedQoSForPod,
+		pluginName:         EvictionPluginNameRssOveruse,
+		metaServer:         metaServer,
+		evictionHelper:     NewEvictionHelper(emitter, metaServer, conf),
+
+		memoryEvictionPluginConfig: conf.MemoryPressureEvictionPluginConfiguration,
+		qosConf:                    conf.QoSConfiguration,
+	}
+}
+
+// RssOveruseEvictionPlugin implements the EvictPlugin interface. It triggers pod eviction based on the rss usage ratio.
+// Once a pod use more rss than the specified threshold, this plugin will evict the pod. Its main goal is to
+// make sure sufficient memory for page cache in some scenarios in which service use page cache to improve performance.
+type RssOveruseEvictionPlugin struct {
+	*process.StopControl
+
+	emitter            metrics.MetricEmitter
+	reclaimedPodFilter func(pod *v1.Pod) (bool, error)
+	pluginName         string
+	metaServer         *metaserver.MetaServer
+	evictionHelper     *EvictionHelper
+
+	memoryEvictionPluginConfig *evictionconfig.MemoryPressureEvictionPluginConfiguration
+	qosConf                    *generic.QoSConfiguration
+}
+
+func (r *RssOveruseEvictionPlugin) Name() string {
+	if r == nil {
+		return ""
+	}
+
+	return r.pluginName
+}
+
+func (r *RssOveruseEvictionPlugin) ThresholdMet(c context.Context) (*pluginapi.ThresholdMetResponse, error) {
+	return &pluginapi.ThresholdMetResponse{
+		MetType: pluginapi.ThresholdMetType_NOT_MET,
+	}, nil
+}
+
+func (r *RssOveruseEvictionPlugin) GetTopEvictionPods(c context.Context, request *pluginapi.GetTopEvictionPodsRequest) (*pluginapi.GetTopEvictionPodsResponse, error) {
+	return &pluginapi.GetTopEvictionPodsResponse{}, nil
+}
+
+func (r *RssOveruseEvictionPlugin) GetEvictPods(c context.Context, request *pluginapi.GetEvictPodsRequest) (*pluginapi.GetEvictPodsResponse, error) {
+	result := make([]*pluginapi.EvictPod, 0)
+
+	if !r.memoryEvictionPluginConfig.DynamicConf.EnableRssOveruseDetection() {
+		return &pluginapi.GetEvictPodsResponse{EvictPods: result}, nil
+	}
+
+	for i := range request.ActivePods {
+		pod := request.ActivePods[i]
+		enabled, threshold := qos.GetRSSOverUseEvictEnabledAndThreshold(r.qosConf, pod)
+		if !enabled {
+			continue
+		}
+
+		if threshold == apiconsts.PodAnnotationMemoryEnhancementRssOverUseThresholdNotSet {
```

Comment:
actually it will, because user might not set threshold explicitly.In this case, `threshold` will be `PodAnnotationMemoryEnhancementRssOverUseThresholdNotSet` too. I'll refactor it since this part of logic is hard to read.